src/Build.UnitTests/FileUtilitiesRegex_Tests.cs

```diff
@@ -0,0 +1,617 @@
+
+        [Fact]
+        public void DrivePatternIsMatchAllProperFormats()
+        {
+            string s;
+            for (char i = 'A'; i <= 'Z'; i++)
+            {
+                s = i + ":";
+                FileUtilitiesRegex.DrivePattern.IsMatch(s).ShouldBeTrue();
+                s = (char)(i + ('a' - 'A')) + ":";
+                FileUtilitiesRegex.DrivePattern.IsMatch(s).ShouldBeTrue();
+            }
+        }
+
+        [Fact]
+        public void IsDrivePatternAllProperFormats()
+        {
+            string s;
+            for (char i = 'A'; i <= 'Z'; i++)
+            {
+                s = i + ":";
+                FileUtilitiesRegex.IsDrivePattern(s).ShouldBeTrue();
+                s = (char)(i + ('a' - 'A')) + ":";
+                FileUtilitiesRegex.IsDrivePattern(s).ShouldBeTrue();
+            }
+        }
+
+        [Fact]
+        public void StartWithDrivePatternIsMatchAllProperFormats()
+        {
+            string s;
+            for (char i = 'A'; i <= 'Z'; i++)
+            {
+                s = i + ":";
+                FileUtilitiesRegex.StartWithDrivePattern.IsMatch(s).ShouldBeTrue();
+                s = (char)(i + ('a' - 'A')) + ":";
+                FileUtilitiesRegex.StartWithDrivePattern.IsMatch(s).ShouldBeTrue();
+            }
+        }
+
+        [Fact]
+        public void DoesStartWithDrivePatternAllProperFormats()
+        {
+            string s;
+            for (char i = 'A'; i <= 'Z'; i++)
+            {
+                s = i + ":";
+                FileUtilitiesRegex.DoesStartWithDrivePattern(s).ShouldBeTrue();
+                s = (char)(i + ('a' - 'A')) + ":";
+                FileUtilitiesRegex.DoesStartWithDrivePattern(s).ShouldBeTrue();
+            }
+        }
+
+        /// <summary>
+        /// Tests DrivePattern regex using various invalid drive patterns.
+        /// </summary>
+        [Theory]
+        [InlineData("C: ")]
+        [InlineData(" C:")]
+        [InlineData(" :")]
+        [InlineData("CC:")]
+        [InlineData("::")]
+        [InlineData("C:/")]
+        [InlineData("C:\\")]
+        [InlineData("C\\")]
+        [InlineData(":/")]
+        [InlineData("  ")]
+        [InlineData("")]
+        public void DrivePatternIsMatchInvalidFormat(string value)
+        {
+            FileUtilitiesRegex.DrivePattern.IsMatch(value).ShouldBe(false);
+        }
+
+
+        /// <summary>
+        /// Tests DrivePattern without regex using various invalid drive patterns.
+        /// </summary>
+        [Theory]
+        [InlineData("C: ")]
+        [InlineData(" C:")]
+        [InlineData(" :")]
+        [InlineData("CC:")]
+        [InlineData("::")]
+        [InlineData("C:/")]
+        [InlineData("C:\\")]
+        [InlineData("C\\")]
+        [InlineData(":/")]
+        [InlineData("  ")]
+        [InlineData("")]
+        public void IsDrivePatternInvalidFormat(string value)
+        {
+            FileUtilitiesRegex.IsDrivePattern(value).ShouldBe(false);
```

Comment:
Now that these are simplified, I'd pull the check-via-legacy-regex and the check-via-new-code calls into single methods for each valid/invalid case. The important thing is that they continue to match and do the expected thing.